music21/romanText/translate.py

```diff
@@ -1605,6 +1606,94 @@ def testTuplets(self):
+
+        def _repeat_tester(
+            repeat_bar: bar.Repeat, direction: str, offset: float, mNumber: int
+        ) -> None:
+            self.assertEqual(repeat_bar.direction, direction)
+            self.assertEqual(repeat_bar.offset, offset)
+            self.assertEqual(repeat_bar.activeSite.number, mNumber)
+
+        def _test_expanded(s: stream.Stream, quarterLength: float) -> None:
+            # NB repeat.Expander does not work on stream, only on part. Is this
+            #   expected behavior?
+            p = next(s[stream.Part])
+            e = repeat.Expander(p)
+            self.assertTrue(e.repeatBarsAreCoherent())
+            p2 = e.process()
+            self.assertEqual(p2.quarterLength, quarterLength)
+
+        # Test simple repeats
+        simple_repeats = '''Time Signature: 3/4
+m1: ||: V
+m2: I :||
+'''
+        s = converter.parse(simple_repeats, format='romanText')
+        start_repeat, end_repeat = s[bar.Repeat]
+        _repeat_tester(start_repeat, 'start', 0.0, 1)
+        _repeat_tester(end_repeat, 'end', 3.0, 2)
+        _test_expanded(s, 12.0)
+
+        single_bar_repeats = '''Time Signature: 2/4
+m1: ||: I :||
+'''
+        s = converter.parse(single_bar_repeats, format='romanText')
+        start_repeat, end_repeat = s[bar.Repeat]
+        _repeat_tester(start_repeat, 'start', 0.0, 1)
+        _repeat_tester(end_repeat, 'end', 2.0, 1)
+        _test_expanded(s, 4.0)
+
+        empty_bars_with_repeats = '''Time Signature: 2/4
+m1: I
+m2: ||:
+m3: :||
+m4: ||: :||'''
+        s = converter.parse(empty_bars_with_repeats, format='romanText')
+        start_repeat1, end_repeat1, start_repeat2, end_repeat2 = s[bar.Repeat]
+        _repeat_tester(start_repeat1, 'start', 0.0, 2)
+        _repeat_tester(end_repeat1, 'end', 2.0, 3)
+        _repeat_tester(start_repeat2, 'start', 0.0, 4)
+        _repeat_tester(end_repeat2, 'end', 2.0, 4)
+        _test_expanded(s, 14.0)
+
+        three_endings = '''Time Signature: 3/4
+m1: ||: I
+m2a: IV :||
+m2b: V :||
+m2c: I'''
+        s = converter.parse(three_endings, format='romanText')
+        start_repeat, end_repeat1, end_repeat2 = s[bar.Repeat]
+        _repeat_tester(start_repeat, 'start', 0.0, 1)
+        _repeat_tester(end_repeat1, 'end', 3.0, 2)
+        _repeat_tester(end_repeat2, 'end', 3.0, 2)
+        first_ending, second_ending, third_ending = s[spanner.RepeatBracket]  # pylint: disable=unused-variable
+        # TODO the following tests don't work; repr(first_ending) gives
+        #   '<music21.spanner.RepeatBracket 1 <music21.stream.Measure 2a offset=3.0>>'
+        #   but first_ending.getNumberList() returns [1]
+        #   What is the correct approach to checking that the measures are as we
+        #   expect? Or should we just forget about it since _test_expanded passes?
+        # self.assertEqual(first_ending.getNumberList(), ['2a'])
+        # self.assertEqual(second_ending.getNumberList(), ['2b'])
+        _test_expanded(s, 18.0)
```

Comment:
I'm not sure that I follow, but can you make an issue if it's something beyond this PR?  It seems like we're not setting numberList properly.

Reply:
I investigated this some more and I realized that RepeatBracket.number or getNumberList indicates not the measure numbers that the bracket spans, but instead the "ending number" (e.g., 1 for the first ending, 2 for the second ending, and so on). I was misunderstanding.

I'll submit a new commit to clean up this misleading comment and implement this test correctly.